plugins/redis/redis.cpp

```diff
@@ -114,21 +126,24 @@ public :
     bool exists(ICodeContext * ctx, const char * key);
 
 protected :
+    void redisSetTimeout();
+    void redisConnectWithTimeout();
+    unsigned timeLeft();
     void parseOptions(ICodeContext * ctx, const char * _options);
-    void connect(ICodeContext * ctx, unsigned __int64 _database, const char * password);
-    void selectDB(ICodeContext * ctx, unsigned __int64 _database);
+    void connect(ICodeContext * ctx, int _database, const char * password);
+    void selectDB(ICodeContext * ctx, int _database);
     void resetContextErr();
     void readReply(Reply * reply);
     void readReplyAndAssert(Reply * reply, const char * msg);
-    void readReplyAndAssertWithKey(Reply * reply, const char * msg, const char * key);
+    void readReplyAndAssertWithCmdMsg(Reply * reply, const char * msg, const char * key = NULL);
     void assertKey(const redisReply * reply, const char * key);
     void assertAuthorization(const redisReply * reply);
     void assertOnError(const redisReply * reply, const char * _msg);
-    void assertOnCommandError(const redisReply * reply, const char * cmd);
-    void assertOnCommandErrorWithDatabase(const redisReply * reply, const char * cmd);
-    void assertOnCommandErrorWithKey(const redisReply * reply, const char * cmd, const char * key);
-    void assertConnection();
-    void updateTimeout(unsigned __int64 _timeout);
+    void assertOnErrorWithCmdMsg(const redisReply * reply, const char * cmd, const char * key = NULL);
+    void assertConnection(const char * _msg);
+    void assertConnectionWithCmdMsg(const char * cmd, const char * key = NULL);
+    void rtlFail(const char * cmd, const char * errmsg, const char * key = NULL);
```

Comment:
fail without the rtl prefix might be clearer.  (rtl stands for run time library.)